client/src/app/core/repositories/polls/poll-repository.service.ts

```diff
@@ -0,0 +1,383 @@
+import { Injectable } from '@angular/core';
+
+import { PollAction } from 'app/core/actions/poll-action';
+import {
+    DEFAULT_FIELDSET,
+    Fieldsets,
+    SimplifiedModelRequest
+} from 'app/core/core-services/model-request-builder.service';
+import { FixedPointNumber, Id } from 'app/core/definitions/key-types';
+import { BaseModel } from 'app/shared/models/base/base-model';
+import { Identifiable } from 'app/shared/models/base/identifiable';
+import { Poll } from 'app/shared/models/poll/poll';
+import { PollMethod, PollState, PollType } from 'app/shared/models/poll/poll-constants';
+import { ViewPoll } from 'app/shared/models/poll/view-poll';
+import { User } from 'app/shared/models/users/user';
+import { toFixedPointNumber } from 'app/shared/utils/to-fixed-point-number';
+import { ViewMeeting } from 'app/site/event-management/models/view-meeting';
+import { BaseRepositoryWithActiveMeeting } from '../base-repository-with-active-meeting';
+import { ModelRequestRepository } from '../model-request-repository';
+import { OptionRepositoryService } from './option-repository.service';
+import { RepositoryServiceCollector } from '../repository-service-collector';
+
+interface AnalogPollVotesValues {
+    votescast: FixedPointNumber;
+    votesvalid: FixedPointNumber;
+    votesinvalid: FixedPointNumber;
+    amount_global_yes: FixedPointNumber;
+    amount_global_no: FixedPointNumber;
+    amount_global_abstain: FixedPointNumber;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class PollRepositoryService
+    extends BaseRepositoryWithActiveMeeting<ViewPoll, Poll>
+    implements ModelRequestRepository {
+    public constructor(repoServiceCollector: RepositoryServiceCollector, private optionRepo: OptionRepositoryService) {
+        super(repoServiceCollector, Poll);
+    }
+
+    public getVerboseName = (plural?: boolean): string => {
+        return plural ? 'Polls' : 'Poll';
+    };
+    public getTitle = (viewModel: ViewPoll): string => {
+        return viewModel.title;
+    };
+
+    public getFieldsets(): Fieldsets<Poll> {
+        const listFieldset: (keyof Poll)[] = [
+            'entitled_group_ids',
+            'state',
+            'title',
+            'type',
+            'pollmethod',
+            'content_object_id'
+        ];
+        const detailFieldset: (keyof Poll)[] = listFieldset.concat(
+            'projection_ids',
+            'voted_ids',
+            'votescast',
+            'votesinvalid',
+            'votesvalid',
+            'option_ids',
+            'majority_method',
+            'onehundred_percent_base',
+            'global_option_id'
+        );
+        return {
+            [DEFAULT_FIELDSET]: detailFieldset,
+            list: listFieldset
+        };
+    }
+
+    public getViewModelListByContentObject(contentObject: BaseModel): ViewPoll[] {
```

Comment:
use `collection: Collection` as the parameter, so one is not restricted to use actual models.